src/adapter/src/coord/sequencer.rs

```diff
@@ -3978,6 +3984,31 @@ impl Coordinator {
         }
         Ok(ops)
     }
+
+    /// Installs the in-memory state for the cluster linked to the specified
+    /// object after a create operation, if such a linked cluster exists.
+    pub(crate) async fn maybe_create_linked_cluster(&mut self, linked_object_id: GlobalId) {
+        if let Some(cluster) = self.catalog.get_linked_cluster(linked_object_id) {
+            let name = cluster.name().to_string();
+            self.create_compute_instance(&name).await;
+        }
+    }
+
+    /// Updates the in-memory state for the replicas of the cluster linked to
+    /// the specified object after an alter operation, if such a linked cluster
+    /// exists.
+    pub(crate) async fn maybe_alter_linked_cluster(&mut self, linked_object_id: GlobalId) {
+        if let Some(cluster) = self.catalog.get_linked_cluster(linked_object_id) {
+            // A linked cluster always has exactly one replica named "linked".
+            // The old replica will have been dropped by `catalog_transact`,
+            // both from the catalog state and from the controller. The new
+            // replica will be in the catalog state, and needs to be recreated
+            // in the controller.
```

Comment:
Are there any asserts we can add here to validate all of this?

Reply:
Hmm, good question. Added the one I could think of, about there being exactly one replica.